app/budget/services/budgetReducers.js

```diff
@@ -425,6 +435,36 @@ budgetApp.service('budgetReducers', function ($rootScope, $log, budgetSelectors)
 				case SELECT_TERM:
 					ui.selectedTerm = action.payload.term;
 					return ui;
+				case TOGGLE_SELECT_LINE_ITEM:
+					var lineItemId = action.payload.lineItem.id;
+					var index = ui.selectedLineItems.indexOf(lineItemId);
+					if (index == -1) {
+						ui.selectedLineItems.push(lineItemId);
+					} else {
+						ui.selectedLineItems.splice(index, 1);
+						ui.areAllLineItemsSelected = false;
+					}
+					return ui;
+				case SELECT_ALL_LINE_ITEMS:
+					action.payload.lineItems.forEach(function(lineItem) {
```

Comment:
Does SELECT_ALL_LINE_ITEMS need to indicate which lineItems are selected? Isn't it all?

Reply:
the uiReducer is intentionally not given access to line item data by default, so it needs to be provided that information when its relevant to the specific action.